clients/vault/src/system.rs

```diff
@@ -821,55 +818,74 @@ impl VaultService {
 		Ok(())
 	}
 
-	async fn maybe_register_vault(
+	async fn register_vault_with_collateral(
+		&self,
+		vault_id: VaultId,
+		collateral_amount: &Option<u128>,
+	) -> Result<(), Error> {
+		if let Some(collateral) = collateral_amount {
+			tracing::info!("[{}] Automatically registering...", vault_id.pretty_print());
+			let free_balance = self
+				.spacewalk_parachain
+				.get_free_balance(vault_id.collateral_currency())
+				.await?;
+			return self
+				.spacewalk_parachain
+				.register_vault(
+					&vault_id,
+					if collateral.gt(&free_balance) {
+						tracing::warn!(
+							"Cannot register with {}, using the available free balance: {}",
+							collateral,
+							free_balance
+						);
+						free_balance
+					} else {
+						*collateral
+					},
+				)
+				.await
+				.map_err(|e| Error::RuntimeError(e))
+		}
+
+		if let Some(_faucet_url) = &self.config.faucet_url {
```

Comment:
We should keep this as an `else if` statement because otherwise we'll always see the last error.


```suggestion
		} else if let Some(_faucet_url) = &self.config.faucet_url {
```

Reply:
done